### PR TITLE
Update Akka.NET source code to support Akka.Remote.Artery port.

### PR DESCRIPTION
**Pull request by Arkatufus**
Related to #4436
Several changes needed in the base Akka.NET source code to support Akka.Remote.Artery port.

### CODE REVIEW

File: src/core/Akka.Remote/UniqueAddress.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+namespace Akka.Remote
+{
+    // ARTERY: Document that UniqueAddress is a copy of the same class from Akka.Cluster.Member
+    // ARTERY: Temporarily use internal for now, so we don't break API approval list
+    /// <summary>
+    /// Member identifier consisting of address and random `uid`.
+    /// The `uid` is needed to be able to distinguish different
+    /// incarnations of a member with same hostname and port.
+    /// </summary>
+    internal class UniqueAddress : IComparable<UniqueAddress>, IEquatable<UniqueAddress>, IComparable
```
Comment (Arkatufus) [inline src/core/Akka.Remote/UniqueAddress.cs:14]: This code duplication is the same as in the scala code, UniqueAddress code is duplicated in Akka.Cluster and Akka.Remote

File: src/core/Akka.Remote/Akka.Remote.csproj [hunk 2]
```diff
@@ -9,6 +9,7 @@
   </PropertyGroup>
   <ItemGroup>
     <EmbeddedResource Include="Configuration\Remote.conf" />
+    <ProjectReference Include="..\Akka.Streams\Akka.Streams.csproj" />
```
Comment (Arkatufus) [inline src/core/Akka.Remote/Akka.Remote.csproj:12]: Akka.Remote.Artery is coupled to Akka.Streams

File: src/core/Akka/Util/Helpers.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Akka.Util
+{
+    public static class Helpers
+    {
+        public static T Requiring<T>(this T obj, Func<T, bool> cond, string msg)
```
Comment (Arkatufus) [inline src/core/Akka/Util/Helpers.cs:9]: This is a helper fluent build pattern function to do an "assert" in fluent manner.
Usage: `myConfig.GetInt("hocon.path.value").Requiring(i => i > 0, "value must be greater than zero.")`

File: src/core/Akka.Remote/Akka.Remote.csproj (hunk 2 above)
Comment (Aaronontheweb) [reply src/core/Akka.Remote/Akka.Remote.csproj:12]: Makes sense

File: src/core/Akka.Remote/UniqueAddress.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/core/Akka.Remote/UniqueAddress.cs:14]: Got it - so we'll eventually move this type from Akka.Cluster to Akka.Remote? Is that the idea?

File: src/core/Akka/Util/Helpers.cs (hunk 3 above)
Comment (Aaronontheweb) [reply src/core/Akka/Util/Helpers.cs:9]: This is meant to be used in the main Akka.NET code base too then I take it? I.e. not just for testing?

File: src/core/Akka/Util/WildcardIndex.cs [hunk 4]
```diff
@@ -7,21 +7,21 @@ namespace Akka.Util
 {
     internal sealed class WildcardIndex<T> : IEquatable<WildcardIndex<T>> where T : class
     {
-        private readonly WildcardTree<T> _wildcardTree;
-        private readonly WildcardTree<T> _doubleWildcardTree;
+        public WildcardTree<T> WildcardTree { get; }
```
Comment (Aaronontheweb) [inline src/core/Akka/Util/WildcardIndex.cs:10]: Is this used for something inside Artery?

Comment (Arkatufus) [reply src/core/Akka/Util/WildcardIndex.cs:10]: yeah, they're accessed directly from one of the classes. I forgot which one...

Comment (Aaronontheweb) [reply src/core/Akka/Util/WildcardIndex.cs:10]: Got it.

**Comment by Aaronontheweb** [comment]
@Arkatufus CI didn't run, but the changes LGTM

**Pull request merged**